src/Silverback.Core/Messaging/Publishing/IPublisher.cs

```diff
@@ -86,10 +87,13 @@ public interface IPublisher
         /// <param name="message">
         ///     The message to be published.
         /// </param>
+        /// <param name="cancellationToken">
+        ///     A <see cref="CancellationToken" /> used to cancel the operation.
+        /// </param>
         /// <returns>
         ///     A <see cref="Task" /> representing the asynchronous operation.
         /// </returns>
-        Task PublishAsync(object message);
+        Task PublishAsync(object message, CancellationToken cancellationToken = default);
```

Comment:
Adding an optional parameter unfortunately introduces a subtle breaking change.

The overload to be invoked is resolved at compile time and at that moment the compiler adds the default values as well. Therefore if you compiled with a version that didn't declare the optional parameter but you run against the later version which does, you will still run into the `MissingMethodException`.

Let's imagine to have a common library that references the current version of Silverback and calls one of these methods. This library is published as nuget and referenced in a .net application. If the application would upgrade to the new version of Silverback with the optional parameter, it would start to throw.

I tried it, to be absolutely sure: `Unhandled exception. System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task Silverback.Messaging.Publishing.IPublisher.PublishAsync(System.Object)'.`

It would be sufficient to recompile the common library, that's true, but it's still a breaking change that I'd prefer to avoid.
The solution is simple (but a bit annoying) and is to add an overload with the extra parameter, instead of using an optional one.

@igorgiovannini if you don't have time or you're not up for it, I will take care of it. 😃